orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/manifest/DeployManifestStage.java

```diff
@@ -58,11 +59,93 @@ public void taskGraph(@Nonnull Stage stage, @Nonnull TaskNode.Builder builder) {
   }
 
   public void afterStages(@Nonnull Stage stage, @Nonnull StageGraphBuilder graph) {
-    DeployManifestContext context = stage.mapTo(DeployManifestContext.class);
-    TrafficManagement trafficManagement = context.getTrafficManagement();
+    TrafficManagement trafficManagement =
+        stage.mapTo(DeployManifestContext.class).getTrafficManagement();
     if (trafficManagement.isEnabled()) {
-      manifestStrategyStagesAdder.addAfterStages(
-          trafficManagement.getOptions().getStrategy(), graph, context);
+      switch (trafficManagement.getOptions().getStrategy()) {
```

Comment:
The logic here is wrong; we should not be deleting old manifests in `RED_BLACK`.  The existing logic is:
Disable and delete for highlander:
https://github.com/spinnaker/orca/blob/12d98dff16f6e02f5529ac22e9389f66b057ecd0/orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/ManifestHighlanderStrategy.java#L33

Disable only for Red/Black:
https://github.com/spinnaker/orca/blob/12d98dff16f6e02f5529ac22e9389f66b057ecd0/orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/ManifestRedBlackStrategy.java#L33

I think somewhere in the refactor this was broken.

Reply:
👍